Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -438,7 +616,111 @@ def send_asset_attributes_request(asset_id: str) -> Dict[str, Any]:
     return res.json()
 
 
-def test_module(client: Client):
+def get_timestamp(timestamp):
+    return time.mktime(timestamp.timetuple())
+
+
+def generate_export_uuid(client: Client, last_run,
+                         severity: List[str]):
+    """
+    Generate a job export uuid in order to fetch vulnerabilities.
+
+    Args:
+        client: Client class object.
+        first_fetch: time to first fetch from.
+        last_run: last run object.
+        severity: severity of the vulnerabilities to include in the export.
+
+    """
+    demisto.info("Getting vulnerabilities export uuid for report.")
+    last_found: float = get_timestamp(arg_to_datetime(ASSETS_FETCH_FROM))   # type: ignore
+
+    export_uuid = client.get_vuln_export_uuid(num_assets=CHUNK_SIZE, last_found=last_found, severity=severity)
+
+    demisto.info(f'vulnerabilities export uuid is {export_uuid}')
+    last_run.update({'vuln_export_uuid': export_uuid})
+
+
+def generate_assets_export_uuid(client: Client, assets_last_run):
+    """
+    Generate a job export uuid in order to fetch assets.
+
+    Args:
+        client: Client class object.
+        first_fetch: time to first fetch assets from.
+        assets_last_run: assets last run object.
+
+    """
+
+    demisto.info("Getting assets export uuid.")
+    fetch_from = round(get_timestamp(arg_to_datetime(ASSETS_FETCH_FROM)))
+
+    export_uuid = client.export_assets_request(fetch_from=fetch_from)
+    demisto.debug(f'assets export uuid is {export_uuid}')
+
+    assets_last_run.update({'assets_export_uuid': export_uuid})
+
+
+def handle_assets_chunks(client: Client, assets_last_run):
+    stored_chunks = assets_last_run.get('assets_available_chunks', [])
+    updated_stored_chunks = stored_chunks.copy()
+    export_uuid = assets_last_run.get('assets_export_uuid')
+    assets = []
+    for chunk_id in stored_chunks[:MAX_CHUNKS_PER_FETCH]:
+        assets.extend(client.download_assets_chunk(export_uuid=export_uuid, chunk_id=chunk_id))
+        # demisto.debug(f"extended assets are: {assets}")
+        updated_stored_chunks.remove(chunk_id)
+    if updated_stored_chunks:
+        assets_last_run.update({'assets_available_chunks': updated_stored_chunks,
+                                'nextTrigger': '0', "type": 1})
+    else:
+        assets_last_run.pop('nextTrigger', None)
+        assets_last_run.pop('type', None)
+        assets_last_run.pop('assets_available_chunks', None)
+        assets_last_run.pop('assets_export_uuid', None)
+    return assets, assets_last_run
+
+
+def try_get_assets_chunks(client: Client, assets_last_run):
```

Comment:
the name here implies this is fail tolerant or its is a get with try catch
I would go for 
get_asset_export_job_status / get_asset_export_call_status  
which calls the client.get_export_assets_status method.
and returns the tuple of status and chunks and continue from there.